api/mock_authentication.py

```diff
@@ -1,77 +0,0 @@
-from nose.tools import set_trace
```

Comment:
I was able to get rid of this whole thing. Tests now use the new `api.simple_authentication` module which is simpler and doesn't have a name conflict with a mock class defined in `test_authenticator.py`.

This module doesn't work in the new system because it really needs to be configured with complicated JSON objects, and that doesn't fit with the web-based configuration we're going towards.

Many of the features in this module were overkill anyway and I've found simpler ways to test the same functionality. See `test_authenticated_patron_can_authenticate_with_expired_credentials` for an example.